### PR TITLE
Fix browser crashing issues on workspace loader.

### PR DESCRIPTION
**Pull request by monaka**
### What does this PR do?

It fixes crashing web browsers by workspace-loader under unstable networks.

### What issues does this PR fix or reference?

#11893 #12402 (and possibly #9295 #12063)

Note: I found similar bugs in Dashboard also. Issues above will be fixed after merged this and next PR.

### CODE REVIEW

File: workspace-loader/src/json-rpc/websocket-client.ts
Comment (vitaliy-guliy) [inline workspace-loader/src/json-rpc/websocket-client.ts]: private async sleep(ms: number): Promise<void>
?

File: workspace-loader/src/json-rpc/websocket-client.ts
Comment (vitaliy-guliy) [inline workspace-loader/src/json-rpc/websocket-client.ts]: `async send(data: any): void`

Comment (monaka) [reply workspace-loader/src/json-rpc/websocket-client.ts]: I guess we can't define `async send(data: any): void` as async function must return a value, AFAIK. But I see what you want. I'll add an information for the return value.

Comment (vitaliy-guliy) [reply workspace-loader/src/json-rpc/websocket-client.ts]: Yes, indeed. It's not possible to define void for async functions. I didn't know that before.

**Review by vparfonov — APPROVED** [review]

File: workspace-loader/src/json-rpc/che-json-rpc-master-api.ts
Comment (monaka) [reply workspace-loader/src/json-rpc/che-json-rpc-master-api.ts]: I see. Removed.

File: workspace-loader/src/json-rpc/websocket-client.ts
Comment (monaka) [reply workspace-loader/src/json-rpc/websocket-client.ts]: I think there is no need to put `async` here. I added some type information.

File: workspace-loader/src/json-rpc/websocket-client.ts
Comment (monaka) [reply workspace-loader/src/json-rpc/websocket-client.ts]: Modified and force-pushed.

**Comment by monaka** [comment]
I fixed all pointed. Could you review again to make sure?

**Review by vitaliy-guliy — APPROVED** [review]
Nice. Thanks for improvement!

**Review by artaleks9 — DISMISSED** [review]
I see the test report (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1427//Selenium_20tests_20report/) shows a big regression, only 24.476% tests success. So _che-qa_ has to investigate what is the reason.

**Comment by artaleks9** [comment]
ci-test

**Comment by artaleks9** [comment]
ci-test

**Comment by monaka** [comment]
I checked both [CI failed](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1436/org.eclipse.che.dockerfiles$che-dockerfiles-theia/console) and [test regressions](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1427//Selenium_20tests_20report/). I can't believe they are caused by this PR. This doesn't touch Selenium and/or Theia.

**Comment by sleshchenko** [comment]
@monaka Thanks for your contribution. Can wait until your PR will be merged.
For some reasons, Theia module build is not stable. It is not definitely your PR issue.
As about test regressions - I believe that they are not caused by your PR but we need to investigate cause and fix it whether it is CI infrastructure stability or seleniums tests.
Thanks again and sorry for issues you faced with `ci-test`.

**Comment by sleshchenko** [comment]
`ci-test`

**Comment by sleshchenko** [comment]
ci-test

**Comment by monaka** [comment]
@sleshchenko No problem :) I just wrote my impression about logs. I'm making similar patches for dashboard now. I'll send it after this was merged.

**Comment by artaleks9** [comment]
ci-test

**Comment by artaleks9** [comment]
Sorry again for issues you faced with _ci-test._

File: workspace-loader/src/index.ts
Comment (benoitf) [inline workspace-loader/src/index.ts]: hello, why token is removed ?

**Comment by monaka** [comment]
ci-build

**Review by vitaliy-guliy — APPROVED** [review]

**Comment by ashumilova** [comment]
ci-build

**Pull request merged**
